api/google-manager/services/google-manager.js

```diff
@@ -56,6 +96,106 @@ class GoogleManager {
     }
   }
 
+  async createCalendar(title) {
+    try {
+      const calendar = await google.calendar({
+        version: 'v3',
+        auth: this.calendarAuth
+      });
+
+      const createdCalendar = await calendar.calendars.insert({
+        requestBody: {
+          summary: title
+        }
+      });
+      return createdCalendar.data;
+    } catch(err) {
+      console.error(`Google Calendar API returned error ${err} when creating calendar`);
+    }
+  }
+
+  async grantAcl(calendarId, email, role) {
+    try {
+      const calendar = await google.calendar({
+        version: 'v3',
+        auth: this.calendarAuth
+      });
+
+      await calendar.acl.insert({
+        calendarId,
+        requestBody: {
+          role: role,
+          scope: {
+            type: role === 'reader' ? 'group' : 'user',
+            value: email
+          }
+        }
+      });
+    } catch(err) {
+      console.error(`Google Calendar API returned error ${err} when granting acl`);
+    }
+  }
+
+  async createEvent(calendarId, title, groupEmail) {
+    try {
+      const calendar = await google.calendar({
+        version: 'v3',
+        auth: this.calendarAuth
+      });
+
+      const createdEvent = await calendar.events.insert({
+        calendarId,
+        conferenceDataVersion: 1,
+        requestBody: {
+          end: {
+            date: '2021-07-04'
+          },
+          start: {
+            date: '2021-07-04'
+          },
+          recurrence: [
+            'RRULE:FREQ=DAILY'
+          ],
+          attendees: [
+            {
+              email: groupEmail,
+              responseStatus: 'accepted'
+            }
+          ],
+          conferenceData: {
+            createRequest: {
+              requestId: uuidv4(),
+              conferenceSolutionKey: {
+                type: 'hangoutsMeet',
+              },
+            }
+          },
+          summary: `${title} Common Room`,
+        }
+      });
+
+      const {id, summary, conferenceData } = createdEvent.data;
+
+      //gets meetingcode from uri
+      const meetingCode = conferenceData.entryPoints[0].uri.split('/')[3];
```

Comment:
Can you put an example of the uri, and what you expect the splits to be?

Reply:
sure!